tasks/configure.yml

```diff
@@ -9,21 +16,49 @@
     mode: 0700
   register: pgdata_dir_exist
 
-- name: PostgreSQL | Ensure the locale is generated
+- name: PostgreSQL | Ensure the locale is generated | Debian
+  sudo: yes
+  locale_gen: name="{{ postgresql_locale }}" state=present
+  when: ansible_os_family == "Debian"
+
+- name: PostgreSQL | Ensure the locale is generated | RedHat
+  sudo: yes
+  command: >
+    localedef -c -i {{ postgresql_locale_parts[0] }} -f {{ postgresql_locale_parts[1] }}
+    {{ postgresql_locale }}
+  changed_when: false
+  when: ansible_os_family == "RedHat"
+
+- name: PostgreSQL | Reset the cluster - drop the existing one | Debian
+  shell: pg_dropcluster --stop {{ postgresql_version }} {{ postgresql_cluster_name }}
   sudo: yes
-  locale_gen: name={{postgresql_locale}} state=present
+  sudo_user: "{{ postgresql_service_user }}"
+  when: ansible_os_family == "Debian" and postgresql_cluster_reset and pgdata_dir_exist.changed
 
-- name: PostgreSQL | Reset the cluster - drop the existing one
-  shell: pg_dropcluster --stop {{postgresql_version}} {{postgresql_cluster_name}}
+- name: PostgreSQL | Reset the cluster - create a new one (with specified encoding and locale) | Debian
+  shell: >
+    pg_createcluster --start --locale {{ postgresql_locale }}
+    -e {{ postgresql_encoding }} -d {{ postgresql_data_directory }}
+    {{ postgresql_version }} {{ postgresql_cluster_name }}
   sudo: yes
   sudo_user: "{{ postgresql_service_user }}"
-  when: postgresql_cluster_reset and pgdata_dir_exist.changed
+  when: ansible_os_family == "Debian" and postgresql_cluster_reset and pgdata_dir_exist.changed
 
-- name: PostgreSQL | Reset the cluster - create a new one (with specified encoding and locale)
-  shell: pg_createcluster --start --locale {{postgresql_locale}} -e {{postgresql_encoding}} -d {{postgresql_data_directory}} {{postgresql_version}} {{postgresql_cluster_name}}
+- name: PostgreSQL | Initialize the database | RedHat
+  command: >
+    {{ postgresql_bin_directory }}/initdb -D {{ postgresql_data_directory }}
+    --locale={{ postgresql_locale }} --encoding={{ postgresql_encoding }}
   sudo: yes
   sudo_user: "{{ postgresql_service_user }}"
-  when: postgresql_cluster_reset and pgdata_dir_exist.changed
+  when: ansible_os_family == "RedHat" and (postgresql_cluster_reset or pgdata_dir_exist.changed)
```

Comment:
This task was not firing for me, hence my failures..., after having manually run `sudo /usr/pgsql-9.3/bin/postgresql93-setup initdb` then restarting postgres it begins to work... the `ansible_os_family == "RedHat"` condition evaluates `True` but something is not working with `(postgresql_cluster_reset or pgdata_dir_exist.changed)` ..